pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -421,3 +394,91 @@ func (kc *KubeConfig) DeleteWorkspace(ctx context.Context, opts *Options, worksp
 
 	return write(opts, fmt.Sprintf("Workspace \"%s\" deleted.\n", workspaceName))
 }
+
+// getOrgClusterNameandBasePath gets the logical cluster name and the base URL for the current
+// workspace.
+func getOrgClusterNameandBasePath(urlPath string, parent bool) (string, string, error) {
+	// get workspace from current server URL and check it point to an org or the root workspace
+	serverURL, err := url.Parse(urlPath)
+	if err != nil {
+		return "", "", err
+	}
+
+	possiblePrefixes := []string{
+		"/clusters/",
+		path.Join(virtualcommandoptions.DefaultRootPathPrefix, "workspaces") + "/",
+	}
+
+	var clusterName, basePath string
+	for _, prefix := range possiblePrefixes {
+		clusterIndex := strings.Index(serverURL.Path, prefix)
+		if clusterIndex < 0 {
+			continue
+		}
+		clusterName = strings.SplitN(serverURL.Path[clusterIndex+len(prefix):], "/", 2)[0]
+		basePath = serverURL.Path[:clusterIndex]
+	}
+
+	if clusterName == "" {
+		return "", basePath, fmt.Errorf("current cluster URL %s is not pointing to a workspace", serverURL)
+	}
+
+	// derive the org workspace to operator on
+	var orgClusterName string
+	if clusterName == tenancyhelpers.RootCluster {
+		orgClusterName = clusterName
+	} else if org, _, err := tenancyhelpers.ParseLogicalClusterName(clusterName); err != nil {
+		return "", "", fmt.Errorf("unable to parse cluster name %s", clusterName)
+	} else if org == "system:" {
+		return "", "", fmt.Errorf("no workspaces are accessible from %s", clusterName)
+	} else if org == tenancyhelpers.RootCluster {
+		if parent {
+			orgClusterName = tenancyhelpers.RootCluster
+		} else {
+			// already in an org workspace
+			orgClusterName = clusterName
+		}
+	} else {
+		// some other workspace, return org cluster name
+		orgClusterName, err = tenancyhelpers.ParentClusterName(clusterName)
+		if err != nil {
+			// should never happen
+			return "", "", fmt.Errorf("unable to derive parent cluster name for %s", clusterName)
+		}
+	}
+	return orgClusterName, basePath, nil
+}
+
+// getCurrentContextName returns the current context from kubeconfig.
+func (kc *KubeConfig) getCurrentContextName(opts *Options) (string, error) {
+	currentContextName := kc.startingConfig.CurrentContext
+
+	kubectlOverrides := opts.KubectlOverrides
+	if kubectlOverrides.CurrentContext != "" {
+		currentContextName = kubectlOverrides.CurrentContext
+	}
+
+	if currentContextName == "" {
+		return "", errors.New("no current context")
+	}
+
+	return currentContextName, nil
+}
+
+// getOrgName returns the name of the organization.
+func (kc *KubeConfig) getOrgName(currentContext string) (orgName string, err error) {
+	if serverInfo, ok := kc.startingConfig.Clusters[currentContext]; ok {
+		orgName, _, err = getOrgClusterNameandBasePath(serverInfo.Server, true)
```

Comment:
hope this is the right way to get the org name. Since its encoded in cluster name, and we already have the logic written to find it, reusing it here.

Reply:
I assume it is good yes.